specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2018-01-22/iotdps.json

```diff
@@ -0,0 +1,1917 @@
+         "description": "Resource group identifier.",
+         "required": true,
+         "type": "string"
+        },
+        {
+         "name": "provisioningServiceName",
+         "in": "path",
+         "description": "The name of the provisioning service.",
+         "required": true,
+         "type": "string"
+        },
+        {
+         "name": "certificateName",
+         "in": "path",
+         "description": "The name of the certificate create or update.",
+         "required": true,
+         "type": "string",
+         "maxLength": 256
+        },
+        {
+         "name": "certificateDescription",
+         "in": "body",
+         "description": "The certificate body.",
+         "required": true,
+         "schema": {
+          "$ref": "#/definitions/CertificateBodyDescription"
+         }
+        },
+        {
+         "name": "If-Match",
+         "in": "header",
+         "required": false,
+         "type": "string",
+         "description": "ETag of the certificate. This is required to update an existing certificate, and ignored while creating a brand new certificate."
+        }
+       ],
+       "responses": {
+        "200": {
+         "description": "If certificate already exist and update was successful, the operation returns HTTP status code of 201 (OK).",
+         "schema": {
+          "$ref": "#/definitions/CertificateResponse"
+         }
+        },
+        "default": {
+         "description": "Default error response.",
+         "schema": {
+          "$ref": "#/definitions/ErrorDetails"
+         }
+        }
+       },
+       "deprecated": false,
+       "produces": [
+        "application/json"
+       ],
+       "consumes": [
+        "application/json"
+       ]
+      },
+      "delete": {
+       "tags": [
+        "DELETE"
+       ],
+       "operationId": "DpsCertificate_Delete",
+       "summary": "Delete the Provisioning Service Certificate.",
+       "description": "Deletes the specified certificate assosciated with the Provisioning Service",
+       "x-ms-examples": {
+           "DPSDeleteCertificate": { "$ref": "./examples/DPSDeleteCertificate.json" }
+       },
+       "consumes": [],
+       "produces": [
+        "application/json"
+       ],
+       "parameters": [
+        {
+         "$ref": "#/parameters/subscriptionId"
+        },
+        {
+         "name": "resourceGroupName",
+         "in": "path",
+         "required": true,
+         "type": "string",
+         "description": "Resource group identifier."
+        },
+        {
+         "name": "If-Match",
+         "in": "header",
+         "required": true,
+         "type": "string",
+         "description": "ETag of the certificate"
+        },
+        {
+         "name": "provisioningServiceName",
+         "in": "path",
+         "required": true,
+         "type": "string",
+         "description": "The name of the provisioning service."
+        },
+        {
+         "name": "certificateName",
```

Comment:
Is there a reason this definition differs from [the one above](https://github.com/Azure/azure-rest-api-specs/pull/2542/files#diff-84c76e046443226d752065433b02e42aR172)?

I imagine they should be the same. To ensure that these and all other instances of it are the same, I'd recommend creating a definition in the `definitions` section and referencing that throughout the document.

Reply:
@kvish any updates here?

Reply:
there is a difference. the cert name we use is a logical name used to query for that certificate. The certificateName in the body is the actual "physical" name that appears in the signed certificate.

Reply:
Thanks for getting back to me on this, thanks for explaining the difference.